File: 문제풀이/프로그래머스_고득점_KIT/해시/완주하지_못한_선수.py
from collections import defaultdict

def solution(participant, completion):
    cnts = defaultdict(int)
    for p in participant:
        cnts[p] += 1
    for c in completion:
        cnts[c] -= 1
        
    for name, cnt in cnts.items():
        if cnt == 1:
            return name

'''
풀이 시간: 3분
접근:
    동명이인이 있을 수 있다.
후기:
    문제 조건을 잘 읽자.
'''

File: CPP/프로그래머스_고득점_KIT/해시/완주하지_못한_선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string, int> sub;

    for (auto& p : participant) {
        sub[p]++;
    }
    
    for (auto& c : completion) {
        sub[c]--;
    }
    
    string answer;
    
    for (auto& s : sub) {
        if (s.second == 1) {
            answer = s.first;
            break;
        }
    }
    
    return answer;
}


/*
풀이 시간: 5분 30초
접근:
    participant에는 있는데 completion에는 없는 선수
    
    동명이인 처리는?
후기:
    문법에 익숙해져가는 중이다.
*/